src/udp/connection/client_id.rs

```diff
@@ -0,0 +1,125 @@
+//! ClientId is a unique ID for the UDP tracker client.
+//! Currently implemented with a hash of the IP and port.
+use std::net::{SocketAddr, IpAddr};
+
+use blake3::OUT_LEN;
+
+#[derive(PartialEq, Debug, Copy, Clone)]
+pub struct ClientId {
+    value: [u8; 4],
+}
+
+impl ClientId {
+    /// It generates the ID from the socket address (IP + port)
+    pub fn from_socket_address(remote_address: &SocketAddr) -> Self {
+        let unique_socket_id = generate_id_for_socket_address(remote_address);
+        ClientId {
+            value: unique_socket_id
+        }
+    }
+
+    /// It generates the ID with a previously generated value
+    pub fn from_slice(slice: &[u8]) -> Self {
+        let mut client_id = ClientId {
+            value: [0u8; 4]
+        };
+        client_id.value.copy_from_slice(slice);
+        client_id
+    }
+
+    pub fn to_bytes(&self) -> [u8; 4] {
```

Comment:
Returning `&[u8; 4]` or even `&[u8]` is preferred over `[u8; 4]`, because borrowing is cheaper than creating a new owned array. Unless we want to consume `self`. In that case it should be:

```rust
pub fn to_bytes(self) -> [u8; 4] {
    self.value
}
```

The ClientId will then be dropped, because `value` is moved.